apps/vr-tests-react-components/src/utilities/getStoryVariant.ts

```diff
@@ -0,0 +1,26 @@
+import { ComponentStory } from '@storybook/react';
+import { parameters } from '@fluentui/react-storybook-addon';
+
+export const DARK_MODE = 'Dark Mode';
+export const HIGH_CONTRAST = 'High Contrast';
+export const RTL = 'RTL';
+
+/**
+ * Helper function that returns a RTL, Dark Mode or High Contrast variant of an existing story.
+ */
+export const getStoryVariant = (
```

Comment:
**suggestion:**

the mixture of variants context is bit unfortunate as it complicates the underlying implementation. We could refactor this though to be more readable/explicit:

```ts
const theme = {
  'Dark Mode': 'web-dark',
  'High Contrast': 'teams-high-contrast',
} as const;

type Variant = typeof DARK_MODE | typeof HIGH_CONTRAST | typeof RTL;

/**
 * Helper function that returns a RTL, Dark Mode or High Contrast variant of an existing story.
 */
export const getStoryVariant = (story: ComponentStory<any>, variant: Variant) => {
  const fluentTheme = getTheme(variant);
  const dir = getDir(variant);

  return {
    ...story,
    render: story,
    storyName: `${getStoryName(story)} - ${variant}`,
    parameters: parameters({ fluentTheme, dir, mode: 'vr-test' }),
  };
};

function getTheme(variant: Variant) {
  if (variant === 'RTL') {
    return 'web-light';
  }
  return theme[variant];
}
function getDir(variant: Variant) {
  if (variant !== 'RTL') {
    return 'rtl';
  }

  return 'ltr';
}

function getStoryName(story: ComponentStory<any>) {
  if (story.storyName) {
    return story;
  }

  return story.name.replace(/([a-z])([A-Z])/g, '$1 $2');
}
```

Reply:
Tests added here: [b469b71](https://github.com/microsoft/fluentui/pull/25108/commits/b469b71efff1456902fb6599981406bde2724d8b)